docs/development/scale-testing.md

```diff
@@ -107,3 +107,28 @@ cd nexodus/hack/e2e-scripts
 ```
 
 The difference is this script adds the self-signed cert from the kind deployment, that is created with the `make ca-cert` in the KIND configuration. The cert has to be in the same directory as the script in a file named `rootCA.pem` in order to be copied to each container for import.
+
+## Nexodus Control Plane Scale Testing Results (as of Oct, 2023)
+
+### High level scale results
+
+- Nexodus Control Plane can scale to 10000 devices with 200 organization with 50 devices per organization.
+- Nexodus Control Plane can scale to 1000 devices per organization.
+
+### Scale testing setup
+
+- Nexodus control plane stack was replicated into a playground namespace in the same cluster.
+- Using [qa-container-scale.sh](../../hack/e2e-scripts/scale/qa-container-scale.sh) script multiple device running next agent were created and connected to the Nexodus control plane.
+- Used AWS EC2 x2large instance to spawn these nexd devices. Each EC2 instance was able to accommodate 1000 containers. New EC2 VM was added as we need to scale more.
+- Helpful scripts to run scale tests are available in [nexodus/hack/e2e-scripts/scale](../../hack/e2e-scripts/scale) directory.
+
+### Major challenges for scale
+
+- High CPU of some key components - ApiServer and ApiProxy. Major contributor to the high CPU was the slow DB queries (such as fetching devices, security groups). Because of slow DB queries, ApiServer was not able to process the requests fast enough and new device onboarding started failing.
```

Comment:
Change to:

```
A major contributor to the high CPU, were the slow DB queries (such as fetching devices, security groups). Because of the slow DB queries, the ApiServer was not able to process the requests fast enough and new device onboarding started failing.
```